File: pidev-3A4-oussama/src/com/esprit/veltun/model/Reclamation.java
package com.esprit.veltun.model;

import java.sql.Date;
import com.esprit.veltun.model.base.BaseEntity;

public class Reclamation extends BaseEntity {
    private int id_reclamation;
    private String object;
    private String description;
    private String status;
    private Date date_reclamation;

    private int rating ;


    public Reclamation() {
    }

    public Reclamation(String object, String description, String status,Date date_reclamation) {
        this.object = object;
        this.description = description;
        this.status = status;
        this.date_reclamation = date_reclamation;
    }

    public Reclamation(int id_reclamation, String object, String description, String status,Date date_reclamation) {

        this.id_reclamation = id_reclamation;
        this.object = object;
        this.description = description;
        this.status = status;
        this.date_reclamation = date_reclamation;
    }



    public int getId_reclamation() {
        return id_reclamation;
    }

    public void setId_reclamation(int id_reclamation) {
        this.id_reclamation = id_reclamation;
    }


    public String getObject() {
        return object;
    }

    public void setObject(String object) {
        this.object = object;
    }
    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }


    public Date getDate_reclamation() {
        return date_reclamation;
    }

    public void setDate_reclamation(Date date_reclamation) {
        this.date_reclamation = date_reclamation;
    }

    @Override
    public String toString() {
        return "reclamation [  id_reclamation=" + id_reclamation + ", object=" + object
                + ", description=" + description + ", status=" + status + ",date_reclamation=" + date_reclamation +"]";
    }

}

File: webProject/src/Entity/Reclamation.php
<?php
namespace App\Entity;

use App\Repository\ReclamationRepository;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;

#[ORM\Entity(repositoryClass: ReclamationRepository::class)]
class Reclamation
{
#[ORM\Id]
#[ORM\GeneratedValue]
#[ORM\Column]
private ?int $idReclamation = null;

#[ORM\Column(length:10 )]
#[Assert\NotBlank(message: 'minumum 5 caractères')]
private ?string $object = null;

#[ORM\Column(type: 'text')]
#[Assert\NotBlank(message: 'Description cannot be blank')]
private ?string $description = null;

#[ORM\Column(length: 20)]
#[Assert\NotBlank(message: 'Status cannot be blank')]
private ?string $status = null;

#[ORM\Column(type: 'datetime')]
private ?\DateTime $dateReclamation = null;

public function getIdReclamation(): ?int
{
return $this->idReclamation;
}

public function setIdReclamation(?int $idReclamation): void
{
$this->idReclamation = $idReclamation;
}

public function getObject(): ?string
{
return $this->object;
}

public function setObject(?string $object): void
{
$this->object = $object;
}

public function getDescription(): ?string
{
return $this->description;
}

public function setDescription(?string $description): void
{
$this->description = $description;
}

public function getStatus(): ?string
{
return $this->status;
}

public function setStatus(?string $status): void
{
$this->status = $status;
}

public function getDateReclamation(): ?\DateTime
{
return $this->dateReclamation;
}

public function setDateReclamation(?\DateTime $dateReclamation): void
{
$this->dateReclamation = $dateReclamation;
}
    /**
     * @var \User
     *
     * @ORM\ManyToOne(targetEntity="User")
     * @ORM\JoinColumns({
     *   @ORM\JoinColumn(name="CIN", referencedColumnName="CIN")
     * })
     */
    private $cin;
    public function getCin(): \User
    {
        return $this->cin;
    }
    public function setCin(\User $cin): Reclamation
    {
        $this->cin = $cin;
        return $this;
    }

}